projects/plugins/jetpack/extensions/blocks/premium-content/premium-content.php

```diff
@@ -37,6 +40,34 @@ function register_block() {
 			),
 		)
 	);
+
+	register_post_meta(
+		'post',
+		META_NAME_CONTAINS_PAID_CONTENT,
+		array(
+			'show_in_rest'  => true,
+			'single'        => true,
+			'type'          => 'boolean',
+			'auth_callback' => function () {
+				return wp_get_current_user()->has_cap( 'edit_posts' );
+			},
+		)
+	);
+
+	// This ensures Jetpack will sync this post meta to WPCOM.
+	add_filter(
+		'jetpack_sync_post_meta_whitelist',
+		function ( $allowed_meta ) {
+			return array_merge(
+				$allowed_meta,
+				array(
+					META_NAME_CONTAINS_PAID_CONTENT,
+				)
+			);
+		}
+	);
```

Comment:
IIRC, we used to do this but added the fields to the default sync file.  Is this something we should do here too?

Reply:
Actually I think there's a dynamic whitelist for some prefixes 🤔.

Reply:
@n3f 

> but added the fields to the default sync file

I thought the same at first. But what we added in the default sync file are the whitelisted custom-post types for coupons and memberships (see [here](https://github.com/Automattic/jetpack/blob/1adad87a1c3d3ba419a7091786b98eb3153c5bd4/projects/packages/sync/src/class-defaults.php#L446)).  

You can see the other metas https://github.com/Automattic/jetpack/pull/36254/files#diff-9164d71f603706b3c6cc58c71ba3f2fbb12b1dba772adca76793f60c24ddbca7R143-R144